src/rapid_pro_to_engagement_db/rapid_pro_archive_client.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+from core_data_modules.logging import Logger
+from temba_client.v2 import Org, Flow, Run, Contact
+
+
+log = Logger(__name__)
+
+
+class RapidProArchiveClient:
+    def __init__(self, archive_dir):
+        """
+        A reimplementation of RapidProClient which operates on a Rapid Pro archive rather than connecting to a
+        production Rapid Pro workspace. Contains only the functions needed to run the Rapid Pro -> engagement db sync.
+
+        :param archive_dir: Path to a Rapid Pro archive directory created by RapidProClient.export_all_data.
+        :type archive_dir: str
+        """
+        self.archive_dir = archive_dir
+
+    def get_workspace_name(self):
+        with open(f"{self.archive_dir}/org.json") as f:
+            org = Org.deserialize(json.load(f))
+        return org.name
+
+    def get_flow_id(self, flow_name):
+        with open(f"{self.archive_dir}/flows.jsonl") as f:
+            flows = [Flow.deserialize(json.loads(d)) for d in f]
+
+        matching_flows = [f for f in flows if f.name == flow_name]
+
+        if len(matching_flows) == 0:
+            available_flow_names = [f.name for f in flows]
+            raise KeyError(f"Requested flow not found on RapidPro (Available flows: {', '.join(available_flow_names)})")
+        if len(matching_flows) > 1:
+            raise KeyError("Non-unique flow name")
+
+        return matching_flows[0].uuid
+
+    def get_raw_runs(self, flow_id, last_modified_after_inclusive=None):
+        log.info(f"Loading raw runs for flow {flow_id}, modified after {last_modified_after_inclusive}, from archives...")
+        with open(f"{self.archive_dir}/runs.jsonl") as f:
+            runs = [Run.deserialize(json.loads(d)) for d in f]
+
+        flow_runs = [r for r in runs if r.flow.uuid == flow_id]
+        if last_modified_after_inclusive is not None:
+            flow_runs = [r for r in runs if r.modified_on >= last_modified_after_inclusive]
+
+        flow_runs.sort(key=lambda r: r.modified_on)
+
+        log.info(f"Returning {len(flow_runs)} runs")
+        return flow_runs
+
+    def update_raw_contacts_with_latest_modified(self, prev_raw_contacts=None, raw_export_log_file=None):
+        log.info(f"Loading contacts from archives...")
+        with open(f"{self.archive_dir}/contacts.jsonl") as f:
+            contacts = [Contact.deserialize(json.loads(d)) for d in f]
+        log.info(f"Loaded {len(contacts)} contacts")
+        return contacts
```

Comment:
can we remove unused params?

Reply:
I'm thinking if we remove prev_raw_contacts if the action will still be updating. You can leave as is since you mentioned this is a reimplementation.

Reply:
Added a comment explaining why they're unused to avoid future confusion.
Ideally Python would have a way for me to say "this is a partial reimplementation of another class" but as far as I'm aware there isn't a way to do that.